### Import Libaries

In [1]:
import numpy as np
import pandas as pd

import pickle
import json

### Load model, scaler, dan fitur

In [2]:
# Load model dan scaler
with open("aria_model.pkl", "rb") as file_1:
    model = pickle.load(file_1)

with open("aria_s_scaler.pkl", "rb") as file_2:
    standard_scaler = pickle.load(file_2)

with open("aria_mm_scaler.pkl", "rb") as file_3:
    minmax_scaler = pickle.load(file_3)

with open("num_col_nd.txt", "r") as file_4:
    num_col_nd = json.load(file_4)

with open("num_col_sd.txt", "r") as file_5:
    num_col_sd = json.load(file_5)

### Load dataset inference

In [3]:
# Load data inference
data_inf = pd.read_excel("dataset/aria_inference_set.xlsx")
data_inf

v1          v2          v3          v4          v5          v6  \
0  555.795044  204.562483  632.219461  541.989336  594.341383  352.076953   
1  468.935422  350.696390  717.073222  558.737233  389.571071  229.284788   
2  401.576015  260.994494  398.234231  519.860805  633.301786  195.062152   
3  534.293085  337.936235  352.532579  343.135279  570.652612  360.542935   
4  470.726763  336.538334  461.182546  559.235599  441.260496  240.346937   
5  405.295767  312.963472  691.138143  380.453846  692.117397  382.075317   

           v7           v8 sample_type  
0  824.797880  4841.922378       lab 1  
1  657.993738  4425.822347       lab 1  
2  846.720129  4717.856130       lab 2  
3  732.063313  4574.904009       lab 2  
4  645.708946  4403.755687       lab 1  
5  823.161168  4573.825094       lab 2

### Scaling feature

In [4]:
# Scaling berdasarkan fitur yang di load
data_inf_nd = data_inf[num_col_nd]
data_inf_sd = data_inf[num_col_sd]

data_inf_nd_scaled = standard_scaler.transform(data_inf_nd)
data_inf_sd_scaled = minmax_scaler.transform(data_inf_sd)

data_inf_final = np.concatenate([data_inf_nd_scaled, data_inf_sd_scaled], axis=1)
data_inf_final

array([[ 2.10698293,  0.15821534,  0.82034899],
       [-1.0458075 ,  1.05566862,  0.51455236],
       [ 2.5213387 ,  0.50478168,  0.7291713 ],
       [ 0.35419279,  0.97730458,  0.62411416],
       [-1.27800424,  0.96871964,  0.49833532],
       [ 2.0760472 ,  0.82393916,  0.62332126]])

### Predict target

In [7]:
# Prediksi dengan model
data_inf_pred = model.predict(data_inf_final)
data_inf_pred = np.round(data_inf_pred, 2)
data_inf_pred

array([4.69, 4.89, 4.68, 4.66, 4.89, 4.68])

In [8]:
# Concat hasil prediksi dengan data inference
data_inf_pred_df = pd.DataFrame(data_inf_pred, columns=["nutrition_pred"])
data_inf_cmp = pd.concat([data_inf, data_inf_pred_df], axis=1)
data_inf_cmp

v1          v2          v3          v4          v5          v6  \
0  555.795044  204.562483  632.219461  541.989336  594.341383  352.076953   
1  468.935422  350.696390  717.073222  558.737233  389.571071  229.284788   
2  401.576015  260.994494  398.234231  519.860805  633.301786  195.062152   
3  534.293085  337.936235  352.532579  343.135279  570.652612  360.542935   
4  470.726763  336.538334  461.182546  559.235599  441.260496  240.346937   
5  405.295767  312.963472  691.138143  380.453846  692.117397  382.075317   

           v7           v8 sample_type  nutrition_pred  
0  824.797880  4841.922378       lab 1            4.69  
1  657.993738  4425.822347       lab 1            4.89  
2  846.720129  4717.856130       lab 2            4.68  
3  732.063313  4574.904009       lab 2            4.66  
4  645.708946  4403.755687       lab 1            4.89  
5  823.161168  4573.825094       lab 2            4.68

Dari 6 data inference diprediksi oleh model rata-rata nutrisi pada tanaman adalah berkisar 4.7